secret_store/src/key_server_set.rs

```diff
@@ -299,14 +357,13 @@ impl CachedContract {
 	}
 
 	fn read_from_registry(&mut self, client: &Client, new_contract_address: Option<Address>) {
-		self.contract = new_contract_address.map(|contract_addr| {
+		if let Some(contract_addr) = new_contract_address {
```

Comment:
This is a change in the logic. It keeps the old address even if you pass `None` as `new_contract_address`

Reply:
great catch!